compiler/noirc_evaluator/src/ssa/ir/types.rs

```diff
@@ -29,15 +31,28 @@ impl NumericType {
         }
     }
 
-    /// Returns true if the given Field value is within the numeric limits
+    /// Returns true if the given BigInt value is within the numeric limits
     /// for the current NumericType.
-    pub(crate) fn value_is_within_limits(self, field: FieldElement) -> bool {
+    pub(crate) fn value_is_within_limits(self, value: &BigInt) -> bool {
         match self {
-            NumericType::Signed { bit_size } | NumericType::Unsigned { bit_size } => {
-                let max = 2u128.pow(bit_size) - 1;
-                field <= max.into()
+            NumericType::Signed { bit_size } => {
+                let min: BigInt = -BigInt::from_i128(1_i128 << (bit_size - 1)).unwrap();
+                let max: BigInt = BigInt::from_i128(1_i128 << (bit_size - 1)).unwrap() - 1;
+                min <= *value && *value <= max
```

Comment:
💭 I wonder if there's a way to do this without allocating the `min` and `max` big ints. `BigInt` can only be compared with other big ints...